onnx/defs/schema.h

```diff
@@ -865,22 +866,39 @@ class OpSchemaRegistry final : public ISchemaRegistry {
       // versining is not meaningful and that domain should have only one
       // version.
       map_[AI_ONNX_PREVIEW_TRAINING_DOMAIN] = std::make_pair(1, 1);
+      // Version corresponding last release of ONNX. Update this to match with
+      // the max version above in a *release* version of ONNX. But in other versions,
+      // the max version may be ahead of the last-release-version.
+      last_release_version_map_[ONNX_DOMAIN] = 12;
+      last_release_version_map_[AI_ONNX_ML_DOMAIN] = 2;
+      last_release_version_map_[AI_ONNX_TRAINING_DOMAIN] = 1;
+      last_release_version_map_[AI_ONNX_PREVIEW_TRAINING_DOMAIN] = 1;
     }
 
     const std::unordered_map<std::string, std::pair<int, int>>& Map() const {
       return map_;
     }
 
+    const std::unordered_map<std::string, int>& LastReleaseVersionMap() const {
+      return last_release_version_map_;
+    }
+
     // Add customized domain to min/max version.
     // Onnx partners are able to use onnx operator schema api to
     // register customized op in their own domain.
     void AddDomainToVersion(
         const std::string& domain,
         int min_version,
-        int max_version) {
+        int max_version,
+        int last_release_version = -1) {
```

Comment:
since this method is used to add custom domains to this map, not sure last_release_version  makes sense here...

Reply:
Isn't it possible that the same scenario arises for custom-domains? Namely, the "last official release" vs "current master branch"? Why expose a different API for onnx-domains and custom-domains? If we can keep them as similar/close to each other as possible, it seems better ... it will avoid unexpected issues later on, I think.

Reply:
We can keep this here... I was just wondering whether this will ever be used... last release version makes sense for onnx domains because when backends pick the domain and version map thay want to be able to distinguish between the released vs yet to be released versions... I dont see how this is applicable for custom domains.

Having said that I agree with your point that keeping the apis as similar as possible is better. Can you add a comment to this method with some description for this last_release_version param.

Reply:
Added comment. My thinking is that if we use "my.custom.domain", it could have its own release cycle independent of ONNX. But it would still make sense to talk about the "last released opset version" of "my.custom.domain".